lib/racmaint.go

```diff
@@ -102,32 +102,47 @@ func racMaintMain(shard int, interval int, cmdLineModuleName string) {
 	binds[0], err = os.Hostname()
 	binds[0] = strings.ToUpper(binds[0])
 	binds[1] = strings.ToUpper(cmdLineModuleName) // */
+	waitTime := time.Second * time.Duration(interval)
+	//First time data loading
+	racMaint(&ctx, shard, db, racSQL, cmdLineModuleName, prev, waitTime/2)
+
+	timeTicker := time.NewTicker(waitTime)
 	for {
-		racMaint(ctx, shard, db, racSQL, cmdLineModuleName, prev)
-		time.Sleep(time.Second * time.Duration(interval))
+		select {
+		case <-ctx.Done():
+			logger.GetLogger().Log(logger.Alert, "Application main context has been closed, so exiting from racmaint data reload.")
+			return
+		case <-timeTicker.C:
```

Comment:
I'm fine with this but why do we need to introduce the timeTicker? I feel the existing code is simpler.